seqr/management/commands/run_postgres_database_backup_tests.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+import mock
+from io import BytesIO
+import datetime
+
+from django.core.management import call_command
+from django.test import TestCase
+
+
+class RunProtgresDatabaseBackupTest(TestCase):
+
+    # Test the command without an argument
+    @mock.patch('seqr.management.commands.run_postgres_database_backup.os')
+    @mock.patch('seqr.management.commands.run_postgres_database_backup.datetime')
+    def test_command_no_argument(self, mock_datetime, mock_os):
+        out = BytesIO()
+
+        mock_os.path.isdir.return_value = False
+        mock_os.environ.get.side_effect = ['db_back_bucket', 'localhost', 'unknown']
+        mock_datetime.datetime.now.return_value = datetime.datetime(2020, 4, 27, 20, 16, 01)
+        call_command('run_postgres_database_backup', stdout = out)
+
+        mock_os.mkdir.assert_called_with('/postgres_backups')
+        calls = [
+            mock.call('DATABASE_BACKUP_BUCKET', 'unknown'),
+            mock.call('POSTGRES_SERVICE_HOSTNAME', 'unknown'),
+            mock.call('DEPLOYMENT_TYPE', 'unknown'),
+        ]
+        mock_os.environ.get.assert_has_calls(calls)
+        self.assertEqual(
+            '=====================================\n' +
+            '======== 2020-04-27__20-16-01 ======= \n' +
+            '=====================================\n' +
+            'Creating directory: /postgres_backups\n' +
+            '/usr/bin/pg_dump -U postgres --host localhost seqrdb | gzip -c - > /postgres_backups/seqrdb_unknown_backup_2020-04-27__20-16-01.txt.gz\n' +
+            '/usr/local/bin/gsutil mv /postgres_backups/seqrdb_unknown_backup_2020-04-27__20-16-01.txt.gz gs://db_back_bucket/postgres/seqrdb_unknown_backup_2020-04-27__20-16-01.txt.gz\n' +
+            '/usr/bin/pg_dump -U postgres --host localhost xwiki | gzip -c - > /postgres_backups/xwiki_unknown_backup_2020-04-27__20-16-01.txt.gz\n' +
+            '/usr/local/bin/gsutil mv /postgres_backups/xwiki_unknown_backup_2020-04-27__20-16-01.txt.gz gs://db_back_bucket/postgres/xwiki_unknown_backup_2020-04-27__20-16-01.txt.gz\n',
+            out.getvalue())
+
+        mock_os.system.assert_called_with(
+            '/usr/local/bin/gsutil mv /postgres_backups/xwiki_unknown_backup_2020-04-27__20-16-01.txt.gz gs://db_back_bucket/postgres/xwiki_unknown_backup_2020-04-27__20-16-01.txt.gz')
+
+    # Test the command with different arguments
+    @mock.patch('seqr.management.commands.run_postgres_database_backup.os')
+    @mock.patch('seqr.management.commands.run_postgres_database_backup.datetime')
+    def test_command_with_arguments(self, mock_datetime, mock_os):
```

Comment:
If all the tests in a module need to mock the same thing, you can put the mock decorators on the class itself instead of on each method. So in this case you would put them on top of `RunProtgresDatabaseBackupTest`, and then the test functions would still get the mocks as arguments the way they do now

Reply:
Thank you for letting me know that.